semantic_conventions/trace/database.yaml

```
@@ -203,12 +203,14 @@ groups:
       - id: statement
         tag: call-level
         type: string
-        requirement_level:
-          conditionally_required: >
-            If applicable and not explicitly disabled via instrumentation configuration.
+        requirement_level: optional
         brief: >
           The database statement being executed.
-        note: The value may be sanitized to exclude sensitive information.
+        note: >
+          Not collected by default; can be enabled via instrumentation configuration.
+          This attribute may contain private or sensitive information.
+          The instrumentation library should support accepting a function that will
```

Maybe "may" instead of "should"?
```suggestion
          The instrumentation library may support accepting a function that will
```

There are other legitimate options other than supplying sanitization function:
- [choosing a strategy from a close set of pre-implemented options](https://github.com/open-telemetry/opentelemetry-specification/issues/3104#issuecomment-1411297008) `"omit" \ "include" \ "obfuscate"`. This is friendly for configuring via file or environment variables where users cannot register a function.
- some libraries already give access to "sanitized" data by default so there might not be need to clean it further.
- instrumentation libraries can choose not to record the db statement at all if it's too complicated or risky.

---

I think `should` here is probably closer to the desired attempt (after all, a `should` is a strong recommendation only).